### PR TITLE
Minor cleanup to L02.

### PR DESCRIPTION
**Pull request by bhamodi**
Changelog:
- Update `computers-are-fast` link to https.
- Minor grammatical touch ups.
- Fixed in-line citations by running `BibTex`.

Before:
screen shot 2018-01-21 at 9 23 16 pm
After:
screen shot 2018-01-21 at 9 23 31 pm

### CODE REVIEW

File: lectures/L02.tex [hunk 1]
```diff
@@ -18,7 +18,7 @@ \section*{Modern Processors}
...
+Clearly there is an area in which frequency scaling was effective. Next year's CPU would have a higher clock speed, and higher clock speed means more cycles per second, and more cycles per second means more work is done in a given second, and that means better performance. Except, we hit the wall: clock speeds stopped getting faster around 2005, stopping at around 3 GHz. Speeding them up beyond this would take, well, more voltage which means more power and more heat, and more heat means higher failure/error rates, and more cooling, and the cooling takes power too, and all that waste heat, well, it will eventually, at the end of this chain, make polar bears sad.
```
Comment (jzarnett) [inline lectures/L02.tex]: Perhaps it's just github that fails to show me the difference, but in this and a few other lines the line shows as changed but I don't see any changes. (I do notice in some others that there are some commas being inserted here or there). Am I overlooking something that has been changed or are these just extraneous changes?

File: lectures/L02.tex [hunk 2]
```diff
@@ -36,7 +36,7 @@ \section*{Modern Processors}
...
+Pipelining: you may have heard a bit about this already, especially so if you have taken ECE~429. To complete an instruction there are five basic steps: (1) fetch the instruction from memory, (2) decode the instruction, (3) fetch needed operands, (4) perform the operation, and (5) write the result. So to do an instruction like \texttt{ADD R1, R2}, we need to fetch the instruction, decode it to figure out what is to be done, read the values from \texttt{R1} and \texttt{R2}, do the addition, and then write the result to \texttt{R1}. Thus, even a simple instruction takes more than one clock cycle, but the good news is that the stages can overlap:
```
Comment (bhamodi) [inline lectures/L02.tex]: https://www.diffchecker.com/tNnOkKBp

File: lectures/L02.tex [hunk 3]
```diff
@@ -46,7 +46,7 @@ \section*{Modern Processors}
...
+The next idea relates to getting items from memory. If we do a load from memory, and we are lucky, it is found in the fastest cache and is available in perhaps 2-3 clock cycles. If we must go to memory, it may be 200-300 cycles. If it is in level 2 or level 3 cache, it will be somewhere in between. The key idea, though, is if we are trying to put something in register \texttt{R7}, it will take time until that value is actually present in that register. The simplest approach is to just wait around until that value has arrived, and then go on to the next instruction. That works, but we can do better.
```
Comment (bhamodi) [inline lectures/L02.tex]: https://www.diffchecker.com/B13690PM

File: lectures/L02.tex [hunk 4]
```diff
@@ -83,32 +83,32 @@ \section*{Modern Processors}
...
+For years Intel was trying to push its Itanium processors (which were so unsuccessful they got the nickname ``Itanic''. Ouch). The goal of these was to find static (compile-time) parallelism: if a machine has infinite registers, can speculate infinitely, etc, the program gets sped up. Run all possibilities in parallel and at the end figure out which is right (wasn't this a Nicolas Cage movie?). Unfortunately, it didn't work out very well because this requires the right kind of program and an super smart compiler. Oh yes, and infinite registers requires infinite space as well as infinite money. So instead the quest has turned to how we can get better performance out of x86\ldots
```
Comment (bhamodi) [inline lectures/L02.tex]: https://www.diffchecker.com/V3FjMcDo

File: lectures/L02.tex [hunk 5]
```diff
@@ -83,32 +83,32 @@ \section*{Modern Processors}
...
+DRAM is, however, not the only kind of memory. There is SRAM (Static RAM) which is fast but expensive, the kind of stuff that goes on the CPU die, and it is six transistors per bit. Compared against DRAM which is much cheaper, but slow: one transistor and one capacitor per bit. Improvements in DRAM have not really improved latency but have improved bandwidth; DDR (Dual Data Rate... not Dance Dance Revolution) means there are two transfers per cycle, but it still takes significant time to get any data out. Additionally, DRAM needs occasional refreshes (capacitors...) so sometimes we have to wait for that.
```
Comment (bhamodi) [inline lectures/L02.tex]: https://www.diffchecker.com/jSMqUmTG

**Comment by patricklam** [comment]
I disagree with the last two, which are rhetorically worse or lead to a
sentence without a verb. (Compare is correct, not compared)

**Comment by bhamodi** [comment]
Alright, reverted the last two changes @patricklam and squashed my commits. Should be good to go now.

**Pull request closed without merging**